clang_build/project.py

```diff
@@ -288,35 +351,80 @@ def _fill_dependency_tree(self):
 
             target.config["dependencies"] = dependency_objs
 
-        # Write dotfile with full dependency graph
+        # Create new dotfile
         if create_dotfile:
             _nx.drawing.nx_pydot.write_dot(self._project_tree, str(self._environment.build_directory / 'dependencies.dot'))
+        self._check_for_circular_dependencies()
```

Comment:
For each generated sub-graph of the entire project you check again. But a few lines further down, the entire structure is already checked.

Reply:
will fix.